### DISCUSSION TITLE
I'm not sure I'm using Astronomy_SearchRiseSet correctly

### DISCUSSION THREAD

**Discussion by meshula** [cosinekitty/astronomy#342] [General]
```
        astro_observer_t observer;
        observer.height = 300;
        observer.latitude = 38;
        observer.longitude = 122;
        astro_time_t time = Astronomy_CurrentTime();
        astro_search_result_t sunrise;
        sunrise  = Astronomy_SearchRiseSet(BODY_SUN,  observer, DIRECTION_RISE, time, 300.0);
        astro_utc_t sunrise_tm = Astronomy_UtcFromTime(sunrise.time);
```

Astronomy_CurrentTime returned on March 29, 2024 around 1030 pm in San Francisco.
```
time	astro_time_t	
ut	double	8854.7371337569566
```

and sunrise was calculated as
```
sunrise_tm	astro_utc_t	
year	int	2024
month	int	3
day	int	30
hour	int	21
minute	int	38
second	double	47.116084098815918
```

and if subtract 7 for UTC offset, that gives me 14, which isn't when the will sun rise, of course. 9pm london is 2pm san francisco, so I'm thinking something is completely wrong in my conception of how to call Astronomy_SearchRiseSet.

**Comment by cosinekitty** [comment]
Hi @meshula and welcome to the project! I think I see the problem here:

```
        observer.longitude = 122;
```

Try changing this to

```
        observer.longitude = -122;
```

Longitudes are positive east of the Prime Meridian, negative to the west of it, so California needs a negative longitude.

I hope this helps!

**Reply by meshula** [reply]
Ahhhh! Thanks! Now I get a result within a few seconds of other sources, and given how coarsely I specified the coordinate, well within tolerance!
